packages/@react-aria/selection/src/useTypeSelect.ts

```diff
@@ -48,7 +48,7 @@ export function useTypeSelect(options: TypeSelectOptions): TypeSelectAria {
 
   let onKeyDown = (e: KeyboardEvent) => {
     let character = getStringForKey(e.key);
-    if (!character || e.ctrlKey || e.metaKey) {
+    if (!character || e.ctrlKey || e.metaKey || selectionManager.isFocusWithinItem) {
```

Comment:
I think this is going to be too restrictive. There are many things that could be placed inside a table cell and receive focus, e.g. a switch, a button, etc. I don't think we'd want to restrict type select in these cases. We need to make this specific to textfields. There's been some talk of an edit mode for tables which may be something we should consider.

Did you try simply stopping propagation on keyboard events inside `useTextField`? Actually that's already the case via `useKeyboard`, but only if you pass in an `onKeyDown`/`onKeyUp` handler. Makes sense in general that we only do that if there's a handler, but for text fields, we should do it always I think since the input itself has default keyboard behavior (not implemented in JS). This should also fix cursor navigation with the arrow keys while editing a textfield inside a table cell as well, just just typeahead.

Reply:
If I remember correctly, the problem here is that `useTypeSelect` has a capturing keydown listener, but lemme try again

Reply:
Only way I can think of making the current solution less restrictive is by checking a combination of `tagName` and `type` (and changing what gets set on `selectionManager` instead of `setFocusWithinItem`) when `focus` fires in `useGridCell` which feels kinda gross. Implementing a edit mode for the table feels a lot cleaner. I can try to implement that in this PR instead or close this and open a new one, thoughts?

Reply:
hmm yeah. I don't really remember why we made it a capturing listener, but seems like it has caused a bunch of issues (e.g. that useSelect issue you fixed). What breaks if we change it to not be a capturing listener?

Reply:
The reason `useTypeSelect` uses capturing listeners is to capture the `space` keydowns that happen in Listbox. If a user types "Item 1" we don't want the `space` keydown to select "Item 1".

Reply:
bump